src/definitions.rs

```diff
@@ -0,0 +1,319 @@
+use crate::{
+    module::FuncInfo, plugin::*, tokenizer::is_valid_function_name, Engine, Module, Scope,
+};
+use core::fmt;
+
+#[cfg(feature = "no_std")]
```

Comment:
Rhai uses the dependent crate `no-std-compat` so you can just add:

```rust
#[cfg(feature = "no_std")]
use std::prelude::v1::*;
```

on top of your file and there is no need to manually import anything in the std prelude.